### PR TITLE
feature/12101-make-some-cell-parts-not-clickable

### PR DESCRIPTION
**Pull request by pascal-brause**
## Description
Makes only these parts of the overview cell in the contact journal clickable, which are NOT generated. This means the visitEncounters (persons, places) and the date header. The remaining cell parts will not be clickable (risk encounters, checkin risk encounters, test entries).

This is only the technical solution, the UI part will follow in a separate PR.

## Link to Jira
Mainstory: https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12006
Subtask: https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12101

## Screenshots
https://user-images.githubusercontent.com/75615785/157483988-739a6956-091d-470d-8d9b-2774b7fdc099.mov

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift [hunk 1]
```diff
@@ -8,8 +8,18 @@ class DiaryOverviewDayTableViewCell: UITableViewCell {
...
+		cellViewModel: DiaryOverviewDayCellModel,
+		didTapClickableView: @escaping () -> Void
+	) {
+		self.didTapClickableView = didTapClickableView
+
 		dateLabel.text = cellViewModel.formattedDate
+		dateLabel.accessibilityIdentifier = String(format: AccessibilityIdentifiers.ContactDiaryInformation.Overview.cellDateHeader, cellViewModel.accessibilityIdentifierIndex)
+		let tapOnDateStackViewRecognizer = UITapGestureRecognizer(target: self, action: #selector(clickableAreaWasTapped))
+		dateStackView.addGestureRecognizer(tapOnDateStackViewRecognizer)
+		dateStackView.addSubview(dateLabel)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift:21]: Isn't the `dateLabel` already added from InterfaceBuilder?

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift [hunk 2]
```diff
@@ -8,8 +8,18 @@ class DiaryOverviewDayTableViewCell: UITableViewCell {
...
+
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift:22]: Lost in space 👾

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewTableViewController.swift [hunk 3]
```diff
@@ -133,7 +123,16 @@ class DiaryOverviewTableViewController: UITableViewController {
...
 		}
-		cell.configure(cellViewModel: viewModel.cellModel(for: indexPath))
+		cell.configure(
+			cellViewModel: viewModel.cellModel(for: indexPath),
+			didTapClickableView: { [weak self] in
+				guard let self = self else {
+					Log.error("Create strong self failed. Could not open day.")
+					return
+				}
+				self.onCellSelection(self.viewModel.day(by: indexPath))
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewTableViewController.swift:133]: I think its unlikely in this case, but the index path of the cell could change after other cells getting removed or rearranged. A solution is to load the current indexPath for the cell like this `let currentIndexPath = self.tableView.indexPath(for: cell)`

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift (hunk 1 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift:21]: You are right. I removed it.

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewTableViewController.swift (hunk 3 above)
Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/DiaryOverviewTableViewController.swift]: Good point. I added it.

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**